### PR TITLE
add dropdown

### PR DESCRIPTION
**Pull request by mazmassa**
We are adding the `<DropDown>` component

Screenshot 2023-05-15 at 11 17 45

### CODE REVIEW

File: src/components/Dropdown/Dropdown.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+              // <FranceFlag className="w-7" /> however by some reason
+              // in storybook is not working as expected. See:
+              // https://github.com/gregberge/svgr/issues/473
+              // In the real code it works fine
+              icon: (
+                <div className="w-7">
+                  <FranceFlag />
+                </div>
+              ),
+              item: "français",
```
Comment (maxime-carabina) [inline src/components/Dropdown/Dropdown.stories.tsx:51]:
```suggestion
              item: "french",
```

File: src/components/Dropdown/Dropdown.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        <Dropdown
+          size={"xs"}
+          options={[
+            {
+              icon: (
+                <div className="w-4">
+                  <FranceFlag />
+                </div>
+              ),
+              item: "français",
```
Comment (maxime-carabina) [inline src/components/Dropdown/Dropdown.stories.tsx:94]:
```suggestion
              item: "french",
```

File: src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    md: {
+      button: "w-full px-6 py-3 h-14 rounded-lg hover:rounded-lg",
+      panel: "w-full rounded-lg hover:rounded-lg",
+      item: "w-full px-6 py-3 h-14 rounded-lg hover:rounded-lg",
+      icon: "pr-2",
+    },
+  };
+
+  const firstObject = options[0];
+  const [toggle, setToggle] = useState(false);
```
Comment (maxime-carabina) [inline src/components/Dropdown/Dropdown.tsx:43]:
```suggestion
  const [toggle, setToggle] = useState<bool>(false);
```

I think it's better to precise the type for each var present in a useState. It's more clearly.

File: src/components/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  const [selected, setSelected] = useState(firstObject);
```
Comment (maxime-carabina) [inline src/components/Dropdown/Dropdown.tsx:44]:
```suggestion
  const [selected, setSelected] = useState<IOption>(firstObject);
```

File: src/components/Dropdown/Dropdown.stories.tsx (hunk 2 above)
Comment (mazmassa) [reply src/components/Dropdown/Dropdown.stories.tsx:94]: Yes, it should be like that. People will recognize the work written.

File: src/components/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (mazmassa) [reply src/components/Dropdown/Dropdown.tsx:44]: True!

Comment (mazmassa) [reply src/components/Dropdown/Dropdown.tsx:44]: Updated!

File: src/components/Dropdown/Dropdown.tsx (hunk 3 above)
Comment (mazmassa) [reply src/components/Dropdown/Dropdown.tsx:43]: Agree! However it does not work like that. We apply when we figure it out. WDYT!?

**Review by maxime-carabina — APPROVED** [review]
LGTM

**Pull request merged**